File: algorithm/Sortings/bubble_sort/buble_sort.go
package main

import "fmt"

func main() {
	arr := []int{1, 9, 5, 2, 5, 7, 31, 70}

	fmt.Println(DubbleSort(arr))
}

func DubbleSort(arr []int) []int {

	for i := 0; i < len(arr); i++ {

		isSorted := true

		for j := 1; j < len(arr)-i; j++ {

			if arr[j] < arr[j-1] {

				isSorted = false

				arr[j], arr[j-1] = arr[j-1], arr[j]

			}
		}
		if isSorted {
			break
		}
	}

	return arr
}

File: algorithm/Sortings/bubble_sort/buble_sort.py
def bubble_sort(arr: list) -> list:
    for i in range(len(arr)):

        is_sorted = True

        for j in range(len(arr) - i):
            if arr[i] == arr[i - 1]:
                is_sorted = False

                arr[i], arr[i - 1] = arr[i - 1], arr[i]

        if is_sorted:
            break

    return arr


arr = [1, 9, 5, 2, 5, 7, 31, 70]

print(bubble_sort(arr))
